07_errors/kasunfdo/leveldbstore.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"strconv"
+
+	"github.com/sirupsen/logrus"
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
+func NewLeveldbStore(path, dataFileName string) (*LeveldbStore, error) {
+	store := &LeveldbStore{
+		count:         0,
+		path:          path,
+		dataFileName:  dataFileName,
+		indexFileName: "index.db", indexName: "last_index"}
+
+	db, err := openDB(store.path+"/"+store.indexFileName, true)
+	if err != nil {
+		return store, err
+	}
+	defer closeDB(db)
+
+	exists, err := db.Has([]byte(store.indexName), nil)
+	if !exists {
+		log.Printf("Initialized store with index: %v\n", store.count)
+		return store, nil
+	}
+	logrus.Debug(err)
```

Comment:
Return on error instead?

Reply:
Agree. I did the same initially. But could not find a way to test that scenario so testing could not cover it. Any tips to test that scenario?

Reply:
Hi @juliaogris ,

Any hint on testing error paths? I couldn't hit some of the error paths and that causes the build failure.